frontend/packages/kubevirt-plugin/src/components/virtualization-overview/permissions-card/PermissionsCardPopover.tsx

```diff
@@ -0,0 +1,94 @@
+import * as React from 'react';
+import { Button, Popover, PopoverPosition } from '@patternfly/react-core';
+import { useTranslation } from 'react-i18next';
+import { ModalBody, ModalTitle } from '@console/internal/components/factory';
+import { GreenCheckCircleIcon, YellowExclamationTriangleIcon } from '@console/shared';
+
+import './permissions-card-popover.scss';
+
+type PermissionItemProps = {
+  task: string;
+  capability: boolean;
+  isLoading: boolean;
+};
+
+type PermissionsCardPopoverProps = {
+  capabilitiesData: { [key: string]: boolean };
+};
+
+const PermissionItem: React.FC<PermissionItemProps> = ({ task, capability, isLoading }) => {
+  const Icon = capability ? GreenCheckCircleIcon : YellowExclamationTriangleIcon;
+
+  return (
+    <div className="kv-overview-permissions-card--popover-item">
+      <div className="kv-overview-permissions-card--popover-item-task">{task}</div>
+      {isLoading ? (
+        <div className="skeleton-inventory" />
+      ) : (
+        <div className="co-dashboard-icon">
+          <Icon />
+        </div>
+      )}
+    </div>
+  );
+};
+
+export const PermissionsCardPopover: React.FC<PermissionsCardPopoverProps> = ({
+  capabilitiesData,
+}) => {
+  const { t } = useTranslation();
+
+  const bodyContent = (
+    <>
+      <ModalTitle>{t('kubevirt-plugin~Permissions')}</ModalTitle>
+      <ModalBody>
+        <div className="kv-overview-permissions-card--popover-table-header">
+          <span>Task</span>
+          <span>Capability</span>
+        </div>
+        <PermissionItem
```

Comment:
consider running PermisisonItem as a loop instead of declaring each :)

Reply:
This gave me the push I needed to clean things up a bit more. Take a look and let me know what you think.